src/libs/antares/study/variable-print-info.cpp

```diff
@@ -186,42 +171,12 @@ void AllVariablesPrintInfo::prepareForSimulation(bool userSelection,
     countSelectedLinkVars();
 }
 
-bool AllVariablesPrintInfo::searchIncrementally_getPrintStatus(std::string var_name) const
-{
-    // Finds out if an output variable is selected for print or not.
-    // The search for the variable in the print info list is incremental :
-    // it resumes where it was left at the previous call.
-    // This function is meant to be called over the whole list of variables,
-    // not to find the print status of one isolated variable.
-    // We want to avoid to search from the start of the print info list at each call.
-
-    resetInfoIterator();
-    for (; it_info != allVarsPrintInfo.end(); it_info++)
-    {
-        if ((*it_info)->name() == var_name)
-        {
-            return (*it_info)->isPrinted();
-        }
-    }
-
-    // This is the case where we have an adequacy-draft variable :
-    // in the case of other study modes, we never get here.
-    resetInfoIterator();
-    return true;
-}
-
 bool AllVariablesPrintInfo::isPrinted(std::string var_name) const
 {
-    // Finds out if an output variable selected for print or not.
-    // The search for a variable starts from the beginning of the variable print info list.
-
-    for (; it_info != allVarsPrintInfo.end(); it_info++)
-    {
-        if ((*it_info)->name() == var_name)
-        {
-            return (*it_info)->isPrinted();
-        }
-    }
+    auto has_name = [&](VariablePrintInfo* v) { return v->name() == var_name; };
+    auto it = std::find_if(allVarsPrintInfo.begin(), allVarsPrintInfo.end(), has_name);
+    if (it != allVarsPrintInfo.end())
+        return (*it)->isPrinted();
```

Comment:
Using [std::any_of](https://en.cppreference.com/w/cpp/algorithm/all_any_none_of) will save you a few lines of code